bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyHandlerFactory.java

```diff
@@ -141,8 +143,12 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         return null;
     }
 
-    public Map<String, ShellyBaseHandler> getThingHandlers() {
-        return deviceListeners;
+    public Map<String, ShellyManagerInterface> getThingHandlers() {
+        Map<String, ShellyManagerInterface> handlers = new TreeMap<>();
+        for (Map.Entry<String, ShellyBaseHandler> bh : deviceListeners.entrySet()) {
+            handlers.put(bh.getKey(), bh.getValue());
+        }
+        return handlers;
```

Comment:
```suggestion
    public Map<String, ShellyManagerInterface> getThingHandlers() {
        return new TreeMap<>(deviceListeners);
```

Reply:
Is this method used anywhere? What's the reason for using a TreeMap?

Reply:
changed to HashMap<>